src/frontend/org/voltdb/calciteadapter/rules/physical/VoltDBPhysicalAggregateRule.java

```diff
@@ -0,0 +1,54 @@
+/* This file is part of VoltDB.
+ * Copyright (C) 2008-2018 VoltDB Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with VoltDB.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.voltdb.calciteadapter.rules.physical;
+
+import org.apache.calcite.plan.RelOptRule;
+import org.apache.calcite.plan.RelOptRuleCall;
+import org.apache.calcite.plan.RelTraitSet;
+import org.apache.calcite.rel.RelNode;
+import org.voltdb.calciteadapter.rel.logical.VoltDBLogicalAggregate;
+import org.voltdb.calciteadapter.rel.logical.VoltDBLogicalRel;
+import org.voltdb.calciteadapter.rel.physical.VoltDBAggregate;
+import org.voltdb.calciteadapter.rel.physical.VoltDBPhysicalRel;
+
+
+public class VoltDBPhysicalAggregateRule extends RelOptRule {
+
+    public static final VoltDBPhysicalAggregateRule INSTANCE = new VoltDBPhysicalAggregateRule();
+
+    private VoltDBPhysicalAggregateRule() {
+        super(operand(VoltDBLogicalAggregate.class, VoltDBLogicalRel.VOLTDB_LOGICAL, any()));
+    }
+
+    @Override
+    public void onMatch(RelOptRuleCall call) {
+        VoltDBLogicalAggregate aggregate = call.rel(0);
```

Comment:
This is the rule where you can choose either hash or serial aggregation, right?

Reply:
Actually, this is simple conversion rule to convert VoltDB Logical (Hash) Aggregate to a corresponding VOltdb Physical(also hash) aggregate. A serial one would have its own VoltDBLogicalSerialAggregate and VoltDBSerialAggregate rels plus matching rules.
But it is still WIP

Reply:
In case I didn't make it clear above, rels that are logical shouldn't be describing the actual algorithms to perform the operation, so a logical aggregate rel should not be thought of as a hash aggregate rel, it's a aggregate rel where the algorithm to be used is as yet undefined.

This seems to be different from how you're thinking of it, maybe we should talk about it.

Reply:
I got you. I agree that hash vs serial vs partial belongs to the physical domain. I will refactor this